beacon-chain/powchain/log_processing.go

```diff
@@ -72,32 +72,56 @@ func (w *Web3Service) ProcessDepositLog(depositLog gethTypes.Log) {
 	if int64(index) <= w.lastReceivedMerkleIndex {
 		return
 	}
-
 	w.lastReceivedMerkleIndex = int64(index)
 
 	// We then decode the deposit input in order to create a deposit object
 	// we can store in our persistent DB.
 	depositInput, err := helpers.DecodeDepositInput(depositData)
 	if err != nil {
-		log.Errorf("Could not decode deposit input  %v", err)
+		log.Errorf("Could not decode deposit input %v", err)
 		return
 	}
+
 	deposit := &pb.Deposit{
 		DepositData:     depositData,
 		MerkleTreeIndex: index,
 	}
+
+	validData := true
+
+	// Make sure duplicates are rejected pre-chainstart.
 	if !w.chainStarted {
-		w.chainStartDeposits = append(w.chainStartDeposits, depositData)
-	} else {
-		w.beaconDB.InsertPendingDeposit(w.ctx, deposit, big.NewInt(int64(depositLog.BlockNumber)))
+		allDeps := w.beaconDB.AllDeposits(w.ctx, nil)
+		for _, dep := range allDeps {
```

Comment:
Is there a better way to do this with O(1) lookup time? 
Otherwise this method is going to take longer as more deposits are added since this is linear time checking the entire list of all deposits.

Reply:
Not sure what can be done around this, I changed it to read from web3service chain starts deposits so it's a bit better now. Only faster way I can think of would be to maintain a mapping of bytes -> bool and marking the pubkey as entered by setting the key to true.